src/commenter.js

```diff
@@ -0,0 +1,164 @@
+  hash.update('\0');
+  hash.update(purpose);
+  hash.update('\0');
+  return hash.digest('hex');
+}
+
+// Compatibility for Node 8.
+if (Symbol.asyncIterator === undefined) {
+  Symbol.asyncIterator = Symbol.for('asyncIterator');
+}
+
+/**
+ * Commenter constructs an Octokit instance for use across multiple calls.
+ */
+export default class Commenter {
+  /**
+   * @param {*} auth Any authorization string or object accepted by octokit.
+   */
+  constructor({ auth }) {
+    this.octokit = new Octokit({ auth });
+  }
+
+  /**
+   * Create a comment.
+   *
+   * @param {string} repository The repository on which to create the comment, in slug form.
+   * @param {number|string} pullRequest The pull request number on GitHub which holds the thread
+   *   where the comment should be created.
+   * @param {string} content The GitHub flavored markdown to put in the comment.
+   * @return {Promise<Object>} An object containing details about the created comment.
+   */
+  async createComment({ repository, pullRequest }, content) {
+    const [owner, repo] = repository.split('/');
+    const { data } = await this.octokit.issues.createComment({
+      owner,
+      repo,
+      issue_number: pullRequest,
+      body: content,
+    });
+    return {
+      action: 'create',
+      link: data.html_url,
+      user: data.user.login,
+    };
+  }
+
+  /**
+   * Update a comment.
+   *
+   * @param {Object} comment The object containing
+   * @param {string} content The GitHub flavored markdown to put in the comment.
+   * @return {Promise<Object>} An object containing details about the created comment.
+   */
+  async updateComment(comment, content) {
+    const url = new URL(comment.url),
+      path = url.pathname.split('/');
+    const [, , owner, repo] = path;
+    const { data } = await this.octokit.issues.updateComment({
+      // In an ideal world, updateComment would accept comment or comment's url or other field and
+      // parse these fields from there.
+      owner,
+      repo,
+      comment_id: path[path.length - 1],
+      body: content,
+    });
+    return {
+      action: 'update',
+      link: data.html_url,
+      user: data.user.login,
+    };
+  }
+
+  /**
+   * Find the comment on the given pull request with the given token.
+   *
+   * @param {string} repository The repository on which to create the comment, in slug form.
+   * @param {number|string} pullRequest The pull request number on GitHub which holds the thread
+   *   where the comment should be created.
+   * @param {string} commentToken The comment token to locate.
+   * @return {Promise<Object>} The comment object from GitHub's API.
+   */
+  async findComment({ repository, pullRequest, commentToken }) {
+    const [owner, repo] = repository.split('/');
+    const options = this.octokit.issues.listComments.endpoint.merge({
+      owner,
+      repo,
+      issue_number: pullRequest,
+    });
+    for await (const page of this.octokit.paginate.iterator(options)) {
```

Comment:
Whoa `for await`!

Reply:
So modern 😎

Reply:
:) it ended up being way less code and way less work than the alternative iteration mechanisms